File: benchmarks/wordfreq/wordfreq.py
import time
parts = ["item-" + str(i) for i in range(200000)]
text = " ".join(parts)

start = time.time()
counts = {"seed": 0}
for w in text.split(" "):
    counts[w] = counts.get(w, 0) + 1
keys = sorted(counts.keys())
elapsed = (time.time() - start) * 1000
print(f"distinct = {len(keys)}")
print(f"Time: {elapsed:.0f}ms")

File: benchmarks/wordfreq/wordfreq.ts
const parts: string[] = [];
for (let i = 0; i < 200000; i++) parts.push("item-" + i);
const text = parts.join(" ");

const start = Date.now();
const counts = new Map<string, number>([["seed", 0]]);
for (const w of text.split(" ")) counts.set(w, (counts.get(w) ?? 0) + 1);
const keys = [...counts.keys()].sort();
const elapsed = Date.now() - start;
console.log(`distinct = ${keys.length}`);
console.log(`Time: ${elapsed}ms`);
